pirania-app/files/www/portal/js/voucher.js

```diff
@@ -35,98 +42,104 @@ function authVoucher () {
+      // debugger
+      if (res && res.result[1] && res.result[1].success) {
+        result.innerHTML = int[lang].success
+        show(result)
+        hide(errorElem)
+        init()
+        const urlTo = window.location.href.split('=')[1]
+        if (urlTo) {
+          window.location.href = `http://${urlTo}`
+        }
+      } else if (res && res.result[1] && !res.result[1].success) {
+        errorElem.innerHTML = int[lang].invalid
+        show(errorElem)
+      } else if (res.error) {
+        console.log(res.error)
+        errorElem.innerHTML = res.error
+        show(errorElem)
+
+        ubusError = true
       }
-    } else if (res && res.result[1] && !res.result[1].success) {
-      errorElem.innerHTML = int[lang].invalid
-      show(errorElem)
-    } else if (res.error) {
-      console.log(res.error)
-      errorElem.innerHTML = res.error
+      voucherElem.value = ''
+    })
+    .catch(err => {
+      console.log('Erro no Ubus', err)
+      errorElem.innerHTML = err
       show(errorElem)
-
       ubusError = true
-    }
-    voucherElem.value = ''
-  })
-  .catch((err) => {
-    console.log('Erro no Ubus', err)
-    errorElem.innerHTML = err
-    show(errorElem)
-    ubusError = true
-  })
-
+    })
 }
 
-function getIp() {
-  window.RTCPeerConnection = window.RTCPeerConnection || window.mozRTCPeerConnection || window.webkitRTCPeerConnection;//compatibility for Firefox and chrome
-  var pc = new RTCPeerConnection({iceServers:[]}), noop = function(){};      
-  pc.createDataChannel('');//create a bogus data channel
-  pc.createOffer(pc.setLocalDescription.bind(pc), noop);// create offer and set local description
-  pc.onicecandidate = function(ice) {
+function getIp () {
+  window.RTCPeerConnection =
+    window.RTCPeerConnection ||
+    window.mozRTCPeerConnection ||
+    window.webkitRTCPeerConnection // compatibility for Firefox and chrome
+  var pc = new RTCPeerConnection({ iceServers: [] })
+
+  var noop = function () {}
+  pc.createDataChannel('') // create a bogus data channel
+  pc.createOffer(pc.setLocalDescription.bind(pc), noop) // create offer and set local description
+  pc.onicecandidate = function (ice) {
     if (ice && ice.candidate && ice.candidate.candidate) {
-      var myIP = /([0-9]{1,3}(\.[0-9]{1,3}){3}|[a-f0-9]{1,4}(:[a-f0-9]{1,4}){7})/.exec(ice.candidate.candidate)[1];
-      pc.onicecandidate = noop;
+      var myIP = /([0-9]{1,3}(\.[0-9]{1,3}){3}|[a-f0-9]{1,4}(:[a-f0-9]{1,4}){7})/.exec(
+        ice.candidate.candidate
+      )[1]
+      pc.onicecandidate = noop
       if (myIP) userIp = myIP
       else {
         fetch('http://thisnode.info/cgi-bin/client_ip')
-        .then(i => {
-          console.log(i)
-          return JSON.parse(i) 
-        })
-        .then((res) => {
-          console.log('Ubus res: ', res)
-          if (res && res.result[1]) {
-            userIp = res.result[1]
-          } else if (res.error) {
-            userIp = res.error
+          .then(i => {
```

Comment:
RECOMMENDATION this block looks repeated, could be generalized

Reply:
:+1: